Tests/YPersistenceTests/DataModel/Mocks/Fruit.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  Fruit.swift
+//  YPersistence
+//
+//  Created by Sahil Saini on 19/04/23.
+//  Copyright © 2023 Y Media Labs. All rights reserved.
+//
+
+import Foundation
+import YPersistence
+
+public struct Fruit: CoreModel, Equatable {
+    public let uid: String
+    public let name: String
+    public let isUploaded: Bool
+    public let wasDeleted: Bool
```

Comment:
I don't know that the "model" needs these. They are used to track what changes need to be uploaded, but that can be strictly for the record and not the model.

Reply:
should I move these to `DataRecord` and update the `GroceryProduct` and `Fruit` accordingly?
So the idea is these will be in entity but not with the model. When some model conform with the `DataRecord` it has to implement these two as well. Right?